phoenix-hive/src/main/java/org/apache/phoenix/hive/util/HiveConnectionUtil.java

```diff
@@ -0,0 +1,161 @@
+
+                final Connection conn =
+                        DriverManager.getConnection(PhoenixRuntime.JDBC_PROTOCOL
+                                + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + quorum
+                                + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + port
+                                + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + znode);
+                String autocommit = configuration.get(HiveConfigurationUtil.AUTOCOMMIT);
+                if (autocommit != null && autocommit.equalsIgnoreCase("true")) {
+                    conn.setAutoCommit(true);
+                } else {
+                    conn.setAutoCommit(false);
+                }
+                connection = conn;
+            } catch (ClassNotFoundException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+        return connection;
+    }
+    
+    /**
+     * Returns the {#link Connection} from Configuration
+     * @param configuration
+     * @return
+     * @throws SQLException
+     */
+    //TODO redundant
+    public static Connection getConnection(final Table tbl) throws SQLException {
+        Preconditions.checkNotNull(tbl);
+        Map<String, String> TblParams = tbl.getParameters();
+        String quorum =
+                TblParams.get(HiveConfigurationUtil.ZOOKEEPER_QUORUM) != null ? TblParams.get(
+                    HiveConfigurationUtil.ZOOKEEPER_QUORUM).trim()
+                        : HiveConfigurationUtil.ZOOKEEPER_QUORUM_DEFAULT;
+        String port =
+                TblParams.get(HiveConfigurationUtil.ZOOKEEPER_PORT) != null ? TblParams.get(
+                    HiveConfigurationUtil.ZOOKEEPER_PORT).trim()
+                        : HiveConfigurationUtil.ZOOKEEPER_PORT_DEFAULT;
+        String znode =
+                TblParams.get(HiveConfigurationUtil.ZOOKEEPER_PARENT) != null ? TblParams.get(
+                    HiveConfigurationUtil.ZOOKEEPER_PARENT).trim()
+                        : HiveConfigurationUtil.ZOOKEEPER_PARENT_DEFAULT;
+        if (!znode.startsWith("/")) {
+        	znode = "/" + znode;
+        }
+        try {
+            Class.forName("org.apache.phoenix.jdbc.PhoenixDriver");
+            final Connection conn =
+                    DriverManager.getConnection((PhoenixRuntime.JDBC_PROTOCOL
+                            + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + quorum
+                            + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + port
+                            + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + znode));
+            String autocommit = TblParams.get(HiveConfigurationUtil.AUTOCOMMIT);
+            if(autocommit!=null && autocommit.equalsIgnoreCase("true")){
+                conn.setAutoCommit(true);
+            }else{
+                conn.setAutoCommit(false);
+            }
+            
+            return conn;
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
```

Comment:
Here too. Please make sure there aren't other occurrences of swallowing exceptions. Just throw a SQLException that wraps the ClassNotFoundException.